webapp/src/dogma/features/file/FileList.tsx

```diff
@@ -11,9 +11,21 @@ export type FileListProps<Data extends object> = {
   data: Data[];
   projectName: string;
   repoName: string;
+  handleCopyApiUrl: Function;
+  handleCopyWebUrl: Function;
+  handleCopyAsCliCommand: Function;
+  handleCopyAsCurlCommand: Function;
 };
 
-const FileList = <Data extends object>({ data, projectName, repoName }: FileListProps<Data>) => {
+const FileList = <Data extends object>({
+  data,
+  projectName,
+  repoName,
+  handleCopyApiUrl,
+  handleCopyWebUrl,
+  handleCopyAsCliCommand,
+  handleCopyAsCurlCommand,
```

Comment:
How about providing this function as an object? `CopySupport`?
And use it like:
```
copySupport.handleApiUrl(...)

```